pandas/core/strings.py

```diff
@@ -2989,33 +2990,34 @@ def rindex(self, sub, start=0, end=None):
     3              sWaPcAsE
     dtype: object
     """)
-    _shared_docs['lower'] = dict(type='lowercase', method='lower', version='')
-    _shared_docs['upper'] = dict(type='uppercase', method='upper', version='')
-    _shared_docs['title'] = dict(type='titlecase', method='title', version='')
-    _shared_docs['capitalize'] = dict(type='be capitalized',
-                                      method='capitalize', version='')
-    _shared_docs['swapcase'] = dict(type='be swapcased', method='swapcase',
-                                    version='')
-    _shared_docs['casefold'] = dict(type='be casefolded', method='casefold',
-                                    version='\n    .. versionadded:: 0.25.0\n')
+    _doc_args = {}  # type: Dict[str, Dict[str, str]]
```

Comment:
you renamed this? does this still work?

Reply:
So to be clear I didn't rename this variable I just created a new one to clarify usage and types. `_shared_docs` still exists, but with str types for both keys and values.

It had mixed usage before because some values were also dicts and were being used to update other strings, so somewhat confusing as to the purpose of the variable and was causing typing to fail since the `%` operator can't have a dict as an operand

Reply:
Here's a screenshot of the output of `lower` locally, which is of the items that was moved to the new variable here:

<img width="1069" alt="image" src="https://user-images.githubusercontent.com/609873/55199659-039c2180-5178-11e9-8e64-fbf1f6ad217c.png">

Reply:
this doesn't make any sense, why did you change the name?

Reply:
To be clear I didn't change the name. `_shared_docs` still exists but now with just a type of `Dict[str, str]` so that it only holds docstrings.

Previously the signature was `Dict[str, Union[Dict[str, str]]]` because it was holding not only docstrings but also dicts to use for substitution in other docstrings, which was rather convoluted and was failing typing. 

Instead of mashing all of those types into one `_shared_docs` variable it is arguably cleaner to have one dict containing docstrings (`_shared_docs`) and one containing substitution parameters (here `_doc_args`).

Reply:
ok i see what you did. can you put a comment to reflect this fact